_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,461 @@
+
+```json
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.trusted_connector_endpoints_regex": [
+            "^https://runtime\\.sagemaker\\..*\\.amazonaws\\.com/.*$",
+            "^https://api\\.openai\\.com/.*$",
+            "^https://api\\.cohere\\.ai/.*$",
+            "^https://bedrock\\..*\\.amazonaws.com/.*$"
+        ]
+    }
+}
+```
+{% include copy-curl.html %}
+
+### Enabling ML nodes
+
+Most connectors require the use of dedicated ML nodes. To make sure you have ML nodes enabled, update the following cluster settings:
+
+```json
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.only_run_on_ml_node": true,
+    }
+}
+```
+{% include copy-curl.html %}
+
+If you are running a remote inference or local model, you can set `"plugins.ml_commons.only_run_on_ml_node"` to `false` and use data nodes instead.
+
+
+### Setting up connector access control
+
+To enable access control on the connector API, use the following cluster setting:
+
+```json
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.connector_access_control_enabled": true
+    }
+}
+```
+{% include copy-curl.html %}
+
+When enabled, the `backend_roles`, `add_all_backend_roles`, or `access_model` options are required in order to use the connector API. If successful, OpenSearch returns the following response:
+
+```json
+{
+  "acknowledged": true,
+  "persistent": {
+    "plugins": {
+      "ml_commons": {
+        "connector_access_control_enabled": "true"
+      }
+    }
+  },
+  "transient": {}
+}
+```
+
+## Creating a connector
+
+You can build connectors in two ways:
+
+1. A **standalone connector**, saved in a connector index, can be reused and shared with multiple remote models but requires access to both the model and the third party being accessed by the connector, such as OpenAI.
+
+2. An **internal connector**, saved in the model index, can only be used with one remote model. Unlike a standalone connector, users only need access to the model itself to access an internal connector because the connection is established inside the model.
+
+## Configuration options
+
+The following configuration options are **required** in order to create a connector. These settings can be used for both standalone and internal connectors.
+
+| Field | Data type | Description |
+| :---  | :--- | :--- |
+| `name` | String | The name of the connector. |
+| `description` | String | A description of the connector. |
+| `version` | Integer | The version of the connector. |
+| `protocol` | String | The protocol for the connection. For AWS services such as Amazon SageMaker and Amazon Bedrock, use `aws_sigv4`. For all other services, use `http`. |
+| `parameter` | JSON array | The default connector parameters, including `endpoint` and `model`. 
```

Comment:
`parameter` -> `parameters`

Type is : Map<String, ?>

Reply:
From security team , we should call out all parameters in this block will be overridable in predict request. User can provide parameter with same in predict request to override the default parameter value defined in connector.